src/main/scala/dibl/PrototypeDiagram.scala

```diff
@@ -0,0 +1,244 @@
+
+    val clones = (for {
+      r <- itemMatrix.indices
+      c <- itemMatrix(r).indices.reverse
+    } yield {
+      val item = itemMatrix(r)(c)
+      val stitch = item.stitch
+      val vectorCode = item.vectorCode.toString.toUpperCase
+      val translate = s"transform='translate(${ c * 10 + 38 },${ r * 10 + 1 })'"
+      val nrOfPairsOut = config.pairsOut(r)(c)
+      val opacity = getOpacity(vectorCode, item.isOpaque)
+      val isActiveNode = opacity == "1" && vectorCode != "-"
+      s"""${ warning(vectorCode, translate, nrOfPairsOut) }
+         |<use ${ events(isActiveNode, item.id) }
+         |  xlink:href='#vc$vectorCode'
+         |  id='svg-r${ r + 1 }-c${ c + 1 }'
+         |  $translate
+         |  style='stroke:${ item.color.getOrElse("#000") };opacity:$opacity;'
+         |><title>$stitch</title>
+         |</use>
+         |${ textInput(isActiveNode, r, c, config) }""".stripMargin
+    }).mkString("\n")
+    embed(clones)
+  }
+
+  private def events(isActive: Boolean, id: String) = {
+    if (isActive)
+      s"data-formid='${ id }' onclick='setStitch(this)'"
+    else ""
+  }
+
+  private def textInput(isActive: Boolean, r: Int, c: Int, config: TilesConfig) = {
+    val item = config.itemMatrix(r)(c)
+    if (isActive)
+      s"""<foreignObject x='${ 19 + c * 10 }' y='${ 970 + r * 10 }' width='4em' height='8'>
+         |  <input name='${ item.id }'
+         |    id='${ item.id }'
+         |    type='text'
+         |    value='${ item.stitch }'
+         |    onchange='showProto()'
+         |  ></input>
+         |</foreignObject>
+         |""".stripMargin
+    else ""
+  }
+
+  private def getOpacity(vectorCode: String, isOpaque: Boolean): String = {
+    vectorCode match {
+      case "-" => "0.05"
+      case _ if isOpaque => "1"
+      case _ => "0.3"
+    }
+  }
+
+  private def warning(vectorCode: String, translate: String, nrOfPairsOut: Int) = {
+    (nrOfPairsOut, vectorCode) match {
+      case (2, _) | (_, "-") => "" // a two-in/two-out stitch or no stitch
+      case _ => s"""<use xlink:href='#oops' $translate style='opacity:0.${ 1 + nrOfPairsOut };'></use>"""
+    }
+  }
+
+  def embed(clones: String): String = {
+    val textProps = """y="979.27722" x="-21.02791" style="font-size:3.3px;font-family:Arial;fill:#000000;stroke:none""""
+    val circleProps = """style="fill:none;stroke-width:1" d="m -18.064645,978.05982 c 0,0.55229 -0.223858,1.05229 -0.585787,1.41422 -0.361929,0.36192 -0.861929,0.58578 -1.414213,0.58578 -0.552284,0 -1.052284,-0.22386 -1.414213,-0.58578 -0.361929,-0.36193 -0.585787,-0.86193 -0.585787,-1.41422 0,-0.55228 0.223858,-1.05228 0.585787,-1.41421 0.361929,-0.36193 0.861929,-0.58579 1.414213,-0.58579 0.552284,0 1.052284,0.22386 1.414213,0.58579 0.361929,0.36193 0.585787,0.86193 0.585787,1.41421 z""""
+    // TODO fix compiler stack overflow for:
+    //  val arrowStyle = """style="fill:none;stroke-width:1.1;stroke:#000;marker-end:url(#Arrow1Mend)""""
```

Comment:
@rvanheest Should you have some time and like this puzzle:

With just `mvn clean install` it compiles unless I activate this line.
Travis gets a [stack overflow](https://travis-ci.org/d-bl/GroundForge/jobs/472944329#L1073) which I can reproduce with:
`mvn clean install -DskipTests=true -Dmaven.javadoc.skip=true -B -V`

Reply:
The long string constants were the problem.